### PR TITLE
Release process automation

### PR DESCRIPTION
**Pull request by asvetlov**
The release is created on git tag pushing.

After linters and tests passed, the deploy stage is starting to wait for release approval.

@pytest-dev/pytest-asyncio-admin can approve. If nobody approved the release in 1 day -- deployment is discarded.
After the approval, we have 15 minutes timer for starting the deployment. During these 15 mins we the deployment task can be cancelled (just in case the deployer missed something).

The new release is pushed on PyPI (using generated PyPI upload token with `pytest-asyncio` scope) and on GitHub (visible as GitHub release).

Tag naming: as usual.
`v0.16`, `v0.17`  etc means final releases.
`v0.17a0` means *0.17 alpha 0* release.
`.post` releases are supported but please avoid them -- Dependabot uses SemVer; SemVer handles it as pre-release. Better to publish the next micro-release instead.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -9,6 +9,38 @@ on:
   workflow_dispatch:
 
 jobs:
+  lint:
```
Comment (seifertm) [inline .github/workflows/main.yml:12]: The logs for the *Build package* step shows the name of the built artifacts is `pytest-asyncio-0.1.dev1+ga5205f2.tar.gz` and `pytest_asyncio-0.1.dev1+ga5205f2-py3-none-any.whl`. As far as I understand these are reused in the `deploy` stage later.

I expected them to be to have a version like `pytest_asyncio-0.16.1+gabcd1234…` not `0.1.dev1`.

The checkout action seems to do a shallow clone with `--no-tags`. Is it possible this prevents setuptools_scm from determining the correct version?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -9,6 +9,38 @@ on:
...
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+      prerelease: ${{ steps.version.outputs.prerelease }}
+    steps:
+      - uses: "actions/checkout@v2"
+      - uses: "actions/setup-python@v2"
+        with:
+          python-version: "3.9"
+
+      - name: "Install poetry, check-wheel-content, and twine"
```
Comment (seifertm) [inline .github/workflows/main.yml:24]: The Poetry part in the description no longer seems to be valid. Could you adjust it?

**Review by Tinche — CHANGES_REQUESTED** [review]
Hm it install poetry in the job?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -51,23 +83,32 @@ jobs:
...
-        run: "python -m build"
-      - name: "List result"
-        run: "ls -l dist"
-      - name: "Check wheel contents"
-        run: "check-wheel-contents dist/*.whl"
-      - name: "Check long_description"
-        run: "python -m twine check dist/*"
+    - name: Checkout
+      uses: actions/checkout@v2.4.0
+    - name: Download disctributions
```
Comment (nicoddemus) [inline .github/workflows/main.yml:94]:
```suggestion
    - name: Download distributions
```

**Comment by asvetlov** [comment]
@Tinche we had it in the workflow previously.
Fixed now.

File: .github/workflows/main.yml (hunk 1 above)
Comment (asvetlov) [reply .github/workflows/main.yml:12]: `fetch-depth: 0` should fix it. Let's see

**Comment by eirnym** [comment]
@asvetlov nice algorithm! I'd like to apply similar to my projects. However, I don't see how timers and confirmations are set in configuration. Could you enlighten me?

**Comment by asvetlov** [comment]
@eirnym  I cannot show you our Settings config, it is available for the project admin only.

Please take a look at [GitHub docs first](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) and feel free to ask if you have a question.

**Comment by asvetlov** [comment]
Merging, I'd like to build an alpha release to test the workflow

**Pull request merged**